hathor/transaction/block.py

```diff
@@ -267,14 +284,8 @@ def verify_checkpoint(self, checkpoints: List[Checkpoint]) -> None:
         assert self.storage is not None
         meta = self.get_metadata()
         # XXX: it's fine to use `in` with NamedTuples
-        if Checkpoint(meta.height, self.hash) in checkpoints:
+        if Checkpoint(meta.get_height(soft=True), self.hash) in checkpoints:
             return
-        # otherwise at least one child must be checkpoint validated
-        for child_tx in map(self.storage.get_transaction, meta.children):
-            if child_tx.get_metadata().validation.is_checkpoint():
-                return
-        raise CheckpointError(f'Invalid new block {self.hash_hex}: expected to reach a checkpoint but none of '
```

Comment:
I guess you shouldn't have removed this line of code. I mean, an exception should be raised here, right?

Reply:
It's actually not that straightforward. Because allowing partial children was removed from this PR, this check has to be adapted so an exception that makes sense is raised. The check will now look at the hash if the height matches. But the other part of the verification is left for a future PR that will have to add a metadata for "partially validated children" (or use some other solution).

Reply:
Moved to another PR.